framework/src/multiapps/FullSolveMultiApp.C

```diff
@@ -123,22 +123,27 @@ FullSolveMultiApp::solveStep(Real /*dt*/, Real /*target_time*/, bool auto_advanc
     if (!getParam<bool>("keep_full_output_history"))
       _apps[i]->getOutputWarehouse().reset();
 
-    bool show = (_fe_problem.verboseMultiApps() ||
-                 _apps[i]->getOutputWarehouse().getOutputs<Console>().size() == 0);
-
     Executioner * ex = _executioners[i];
     ex->execute();
-    if (!ex->lastSolveConverged())
-    {
-      last_solve_converged = false;
-      if (show)
-        _console << COLOR_RED << "Subapp " << _apps[i]->name() << " solve Did NOT Converge!"
-                 << COLOR_DEFAULT << std::endl;
-    }
-    else if (show)
-      _console << COLOR_GREEN << "Subapp " << _apps[i]->name() << " solve converged!"
-               << COLOR_DEFAULT << std::endl;
+
+    last_solve_converged = last_solve_converged && ex->lastSolveConverged();
+
+    showStatusMessage(i);
   }
 
   return last_solve_converged || _ignore_diverge;
 }
+
+void
+FullSolveMultiApp::showStatusMessage(unsigned int i) const
+{
+  if (!_fe_problem.verboseMultiApps() &&
+      _apps[i]->getOutputWarehouse().getOutputs<Console>().size() > 0)
+    return;
+  else if (!_executioners[i]->lastSolveConverged())
+    _console << COLOR_RED << "Subapp " << _apps[i]->name() << " solve Did NOT Converge!"
```

Comment:
I'm guessing because when there is a console output, the convergence of the sub-app is obvious, so there is not need to repeat the message, unless we are in verbose mode.